resources/kubernetes/operator/helm/values.yaml

```diff
@@ -4,21 +4,24 @@ opentelemetry-operator:
       - --enable-go-instrumentation
   admissionWebhooks:
     certManager:
-      enabled: false
+      enabled: false # Disable cert-manager for managing admission webhook certificates.
+                     # For production environments, it is recommended to use cert-manager for better security and scalability.
+                     # Refer to: https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-operator#tls-certificate-requirement
+
 
 autoGenerateCert:
-  enabled: true
-  recreate: true
+  enabled: true # Enable/disable automatic certificate generation. Set to false if manually managing certificates.
+  recreate: true # Force certificate regeneration on updates. Only applicable if autoGenerateCert.enabled is true.
 
 crds:
-  create: true
+  create: true # Install the OpenTelemetry Operator CRDs.
 
 defaultCRConfig:
   image:
     repository: "docker.elastic.co/beats/elastic-agent"
     tag: "8.16.0-SNAPSHOT"
   targetAllocator:
-    enabled: false
+    enabled: false # Enable/disable the Operator's Target allocator to decouple the service discovery and metric collection functions of Prometheus. If enabled, per-node allocation strategy must be used with this deployment. Refer to: https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator
```

Comment:
We should hyperlink URLs to reduce length, these will also work in some IDE like VS code.

Refer to [Target Allocator](https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator)